lib/modules/manager/gitlabci/extract.ts

```diff
@@ -165,43 +165,54 @@ export function extractPackageFile(
   let deps: PackageDependency[] = [];
   try {
     // TODO: use schema (#9610)
-    const doc = parseSingleYaml<GitlabPipeline>(replaceReferenceTags(content), {
-      json: true,
-    });
-    if (is.object(doc)) {
-      for (const [property, value] of Object.entries(doc)) {
-        switch (property) {
-          case 'image':
-            {
-              const dep = extractFromImage(
-                value as Image,
-                config.registryAliases,
-              );
-              if (dep) {
-                deps.push(dep);
+    const docs = parseYaml<GitlabPipeline>(
```

Comment:
As we use types here, do you have considered using schemas?

Reply:
There is already issue for that, so I thought do do that in another PR.

Reply:
No it is fine for now.